ops/pebble.py

```diff
@@ -563,6 +576,10 @@ def __str__(self):
         return message
 
 
+class IdentityError(Error):
```

Comment:
I see your note about why you added this, but realistically the server will never return a null identity, so let's remove this error and just call from_dict and let it raise whatever it raises.